src/array/mod.rs

```diff
@@ -72,7 +72,7 @@ pub trait Array: Sized + Send + Sync + 'static {
     type Builder: ArrayBuilder<Array = Self>;
 
     /// Type of element in the array.
-    type Item: ToOwned + ?Sized;
+    type Item: ToOwned + ?Sized + PartialEq;
```

Comment:
I'd prefer write the `PartialEq` bound on where we need, instead of on all items. For example,

```
impl<A, B> RleBlockBuilder<A, B>
 where
     A: Array,
     B: BlockBuilder<A>,
     A::Item: PartialEq,
```